src/Core/src/Platform/Windows/ViewExtensions.cs

```diff
@@ -26,8 +26,17 @@ public static void TryMoveFocus(this FrameworkElement platformView, FocusNavigat
 				FocusManager.TryMoveFocus(direction, new FindNextElementOptions { SearchRoot = elem });
 		}
 
-		public static void UpdateIsEnabled(this FrameworkElement platformView, IView view) =>
-			(platformView as Control)?.UpdateIsEnabled(view.IsEnabled);
+		public static void UpdateIsEnabled(this FrameworkElement platformView, IView view)
+		{
+			if (platformView is Control control)
+				control.UpdateIsEnabled(view);
+			else
+			{
+				platformView.IsHitTestVisible = view.IsEnabled && !view.InputTransparent;
+
+				(view as ILayout)?.InvalidateChildrenIsEnabled();
```

Comment:
Should this be inside the else block? What happens if Control is also ILayout? If not, then it is a no-op, but if it is...